encrypted-media-respec.html

```diff
@@ -1604,15 +1611,95 @@ <h2><dfn>MediaKeys</dfn> Interface</h2>
               The type of session to create. The session type affects the behavior of the returned object.
-            </td></tr></tbody></table><div><em>Return type: </em><code>MediaKeySession</code></div></dd><dt><dfn><code>setServerCertificate</code></dfn></dt><dd>
+            </td></tr></tbody></table><div><em>Return type: </em><code>MediaKeySession</code></div></dd>
+            <dt><dfn><code>getStatusForPolicy</code></dfn></dt>
+            <dd>
+              <p id="status-for-policy">Returns the <a>MediaKeyStatus</a> for a given <a>MediaKeysPolicy</a>.</p>
+
+              <ol class="method-algorithm">
+                <li>If <var>policy</var> has no <a def-id="present-dictionary-member"></a> <a def-id="dictionary-member">members</a>, return a promise rejected with a newly created <a def-id="TypeError"></a>.</li>
+                <li><p>Let <var>promise</var> be a new promise.</p>
+                  <ol>
+                    <li><p>For each <a def-id="dictionary-member">member</a> of <var>policy</var>, run the following steps:</p>
+                      <ol>
+                        <li><p>If the CDM cannot determine the <a>MediaKeyStatus</a> for <a def-id="dictionary-member">member</a>, then reject <var>promise</var> with <a def-id="NotSupportedError"></a> and abort these steps.</p></li>
+                      </ol>
+                    </li>
+                    <li><p>For each <a def-id="dictionary-member">member</a> of <var>policy</var>, run the following steps:</p>
+                      <ol>
+                        <li><p>If the CDM would block presentation of decrypted media data for <a def-id="dictionary-member">member</a>, then resolve <var>promise</var> with <a def-id="status-output-restricted"></a>.</p></li>
+                      </ol>
+                    </li>
+                    <li><p>Resolve <var>promise</var> with <a def-id="status-usable"></a>.</p></li>
+                  </ol>
+                </li>
+                <li><p>Return <var>promise</var>.</p></li>
+              </ol>
+              <table class="parameters">
+                <tbody>
+                  <tr>
+                    <th>Parameter</th>
+                    <th>Type</th>
+                    <th>Nullable</th>
+                    <th>Optional</th>
+                    <th>Description</th>
+                  </tr>
+                  <tr>
+                    <td class="prmName">policy</td>
+                    <td class="prmType">
+                      <code>MediaKeysPolicy</code>
+                    </td>
+                    <td class="prmNullFalse">
+                      <span role="img" aria-label="True">✔</span>
+                    </td>
+                    <td class="prmOptFalse">
+                      <span role="img" aria-label="True">✔</span>
```

Comment:
Why do we allow the `policy` to be `Nullable` and `Optional`? It seems simpler not to allow them. If we do allow, the spec should handle these cases.

Reply:
IIUC, the original idea for this method was to make it extensible, by adding other policies to MediaKeysPolicy. If this is where the API is going then it doesn't make sense to make `minHdcpVersion` required. And dictionary argument must be optional if it has no required fields. I don't think it makes sense to make `minHdcpVersion` required now, if it might not be in the future (however unlikely).

Reply:
Fully agree with what you said! However, the question was about `policy`, not `minHdcpVersion`. The current draft says the `policy` is `Nullable` and `Optional`, and I was asking about that.

Reply:
I actually responded to that. When a dictionary (i.e. the policy) has no required fields then it must be optional/nullable. Respec gives a WebIDL validation error when policy is not optional/nullable.
> And dictionary argument must be optional if it has no required fields.

If you look at [Encrypted Media: HDCP Policy Check](https://wicg.github.io/hdcp-detection/), you will see this is defined the same way. I'm pretty sure that's why @joeyparrish wrote the explainer that way.

Reply:
Thanks for the explanation! I leaned something today. It feels odd to allow calling this method with null, but I guess you could always call it with an empty dictionary, which is the same thing.

@tidoust to double check on this.

Since the parameter is nullable, in the algorithm, should we handle it?

Reply:
The Web IDL definition looks good to me.

For `optional`, that is indeed required (with a default value).

I don't understand what makes the parameter nullable, though? For it to be nullable, [according to Web IDL](https://webidl.spec.whatwg.org/#idl-nullable-type), it would have to be defined with a `?`, as in:

```
Promise<MediaKeyStatus> getStatusForPolicy (
  optional MediaKeysPolicy? policy = {}
);
```

... and Web IDL also says that dictionary types cannot be nullable when used as the type of an operation argument, so that does not seem allowed in any case.

Reply:
I changed `policy` to NOT `Nullable`.